src/Illuminate/Foundation/Application.php

```diff
@@ -552,6 +558,15 @@ protected function markAsRegistered($provider)
 		$this->serviceProviders[] = $provider;
 
 		$this->loadedProviders[$class] = true;
+		
+		$aliases = class_parents($class) + class_implements($class) + array($class);
```

Comment:
It would improve performance for php 5.4 systems. The line length doesn't bother me. Once we get over 120 we should be worried about that.